omnibus/files/private-chef-cookbooks/private-chef/templates/default/elasticsearch.yml.erb

```diff
@@ -18,6 +18,15 @@ network.host: "<%= @listen %>"
 #            http://www.elastic.co/guide/en/elasticsearch/reference/current/modules-discovery.html
 http.port: <%= @port %>
 
+<%
+  # ES7 mandatory configuration
+  if node['private_chef']['elasticsearch']['es_version'].to_f > 7.0
+%>
+discovery.type: single-node
+<%
```

Comment:
on the similar lines there are changes in jvm opts (elasticsearch_jvm.opts.erb) fir elasticsearch 7. I wonder if it might be useful to add a separate templates for es7 and have the recipe pick the correct template based on the version of es.

Reply:
It might be helpful to include context for this in comments. These setting are applicable only for the users using standalone or tiered configurations. For BYOD(external elasticsearch) none of the omnibus configs are applicable.

Reply:
Looks like chef-server lets the user set the listen address for elasticsearch: https://github.com/chef/chef-server/blob/master/omnibus/files/private-chef-cookbooks/private-chef/attributes/default.rb#L170 which is used in the elasticsearch.yml. So in theory this could be a multicluster setup - although I do not think that is something we advertise / support at this point.  It might be worth making a note in the docs that cluster for es will not be supported, if they want to use a cluster they can setup external es.

Reply:
Sure, I will add comments and update docs.

Reply:
Are there any change to elasticsearch_jvm.opts?